src/main/java/uk/gov/justice/hmpps/probationteams/controllers/LocalDeliveryUnitController.java

```diff
@@ -0,0 +1,92 @@
+package uk.gov.justice.hmpps.probationteams.controllers;
+
+import com.microsoft.applicationinsights.TelemetryClient;
+import io.swagger.annotations.*;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import uk.gov.justice.hmpps.probationteams.config.SecurityUserContext;
+import uk.gov.justice.hmpps.probationteams.dto.ErrorResponse;
+import uk.gov.justice.hmpps.probationteams.dto.LocalDeliveryUnitDto;
+import uk.gov.justice.hmpps.probationteams.model.LocalDeliveryUnit;
+import uk.gov.justice.hmpps.probationteams.services.LocalDeliveryUnitService;
+
+import javax.validation.constraints.NotNull;
+
+import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
+
+@Api(tags = {"local-delivery-unit"})
+@RestController
+@RequestMapping(
+        value = "local-delivery-unit",
+        produces = APPLICATION_JSON_VALUE)
+@Slf4j
+@AllArgsConstructor
+public class LocalDeliveryUnitController {
+
+    private final LocalDeliveryUnitService localDeliveryUnitService;
+    private final TelemetryClient telemetryClient;
+    private final SecurityUserContext securityUserContext;
+
+    @GetMapping("/{localDeliveryUnitCode}")
+    @ResponseBody
+    @ApiOperation(value = "Retrieve a Local Delivery Unit",
+            nickname = "Retrieve a Local Delivery Unit")
+    @ApiResponses({
+            @ApiResponse(code = 404, message = "Local Delivery Unit not found"),
+            @ApiResponse(code = 200, message = "OK", response = LocalDeliveryUnit.class)})
+    public ResponseEntity<LocalDeliveryUnitDto> getProbationTeam(
```

Comment:
don't need `ResponseEntity` - worth just returning`LocalDeliveryUnitDto`

Reply:
The controller can return a LocalDeliveryUnitDto or 404 Not Found.
If you look at line 43, localDeliveryUnitService.getLocalDeliveryUnit returns an Optional<LocalDeliveryUnit>

Reply:
the new pattern we're trying to adopt is to then throw an exception and let the global exception handler pick it up and return the 404